docs/developers/guides/gas/gas-fees.mdx

```diff
@@ -36,12 +36,67 @@ and gas used.
 
-Linea also supports [`eth_estimateGas`](https://docs.infura.io/api/networks/linea/json-rpc-methods/eth_estimategas),
-[`eth_gasPrice`](https://docs.infura.io/api/networks/linea/json-rpc-methods/eth_gasprice), and
-[`eth_feeHistory`](https://docs.infura.io/api/networks/linea/json-rpc-methods/eth_feehistory).
+Linea also supports:
+- [`eth_estimateGas`](https://docs.infura.io/api/networks/linea/json-rpc-methods/eth_estimategas)
+- [`eth_gasPrice`](https://docs.infura.io/api/networks/linea/json-rpc-methods/eth_gasprice)
+- [`eth_feeHistory`](https://docs.infura.io/api/networks/linea/json-rpc-methods/eth_feehistory).
+
+## Gas pricing
+
+The gas price you retrieve using methods such as `linea_estimateGas` is based on values calculated
+by the [coordinator](../../../architecture/stack/coordinator/index.mdx), and uses the gas price of 
+the previous block with a profitability multiplier applied.
+
+Firstly, the coordinator records:
+- A fixed cost of 0.03 Gwei, which reflects infrastructure costs, and;
+- Variable cost, which is the cost per byte of data submitted to L1. 
+
+These values are sent to the sequencer so that it can compute the gas prices for the upcoming 
+block and ensure that included transactions are profitable.
+
+Variable cost is calculated with the following formula:
+
+```python
+VARIABLE_COST (4 bytes) = min(max(
+(
+  (
+    ((averageWeightedBaseFee + averageWeigthedPriorityFee) * 
+    blob-submission-expected-execution-gas + averageWeightedBlobBaseFee * expected-blob-gas
+  ) / bytes-per-data-submission) * profit-margin
+)
+, min-bound), max-bound)
+```
+
+The `profit-margin` is `5`, ensuring sustainable network profitability. `min-bound` is `0.050000000` 
+Gwei, and `max-bound` is `0.127500000` Gwei, and together guarantee the gas price stays within a 
+reasonable range.
+
+The variable cost formula is similar to the standard Ethereum gas price formula that Linea uses, 
+except that it allows `linea_estimateGas` to price according to the variable costs of submitting 
+blob data to L1. The amount of the blob data in each block stays roughly consistent, though the 
+amount _per transaction_ varies, so the `linea_estimateGas` API accounts for this, and ensures a 
+gas price is returned for the transaction that reflects the amount of data it contains. 
+
+To determine the priority fee per gas, `linea_estimateGas` takes the previous block's `VARIABLE_COST` 
+into account:
+```python
+min-gas-price = previousBlock.extraData.variable_cost
+baseFeePerGas = vanillaProtocolBaseFee
+priorityFeePerGas = MINIMUM_MARGIN * (min-gas-price * L2_compressed_tx_size_in_bytes / L2_tx_gas_used + extraData.fixed_cost)
+```
 
-You can use `eth_gasPrice` or `eth_feeHistory` to get the gas price, in wei, and you can use 
-`eth_estimateGas` to find out how many units of gas a specific transaction will need. 
+Where:
+- `extraData.variable_cost` is the way the coordinator stores the `VARIABLE_COST` for the previous 
```

Comment:
ExtraData is not some special way. ExtraData is a free real estate, provided by Ethereum protocol. It's 32 bytes that could be anything and it's there in every block. We're using this space to distribute data about fees across Linea network.

Notion of Coordinator doesn't help the understanding of where variable cost comes from and for end user the knowledge that it comes from Sequencer when a block is built is enough